Library/SRValidator.m

```diff
@@ -211,23 +215,20 @@ - (NSString *)SR_path
     NSMenuItem *currentMenuItem = self;
     NSUInteger i = 0;
 
-    do
-    {
+    do {
         [items insertObject:currentMenuItem atIndex:0];
         currentMenuItem = currentMenuItem.parentItem;
         ++i;
     }
     while (currentMenuItem && i < Limit);
 
-    NSMutableString *path = [NSMutableString string];
-
-    for (NSMenuItem *menuItem in items)
-        [path appendFormat:@"%@➝", menuItem.title];
-
-    if ([path length] > 1)
-        [path deleteCharactersInRange:NSMakeRange([path length] - 1, 1)];
+	NSMutableArray *titles = [[NSMutableArray alloc] initWithCapacity:items.count];
+    for (NSMenuItem *menuItem in items) {
+		NSString *title = menuItem.title;
+		if (title) [titles addObject:title];
+	}
 
-    return path;
+    return [titles componentsJoinedByString:@" → "];
```

Comment:
I believe I've seen system UI to use "a➝b", not "a → b". But cannot recall where.